.github/workflows/upgrade.yml

```
@@ -0,0 +1,58 @@
+name: Upgrade
+on:
+  workflow_dispatch:
+      inputs:
+      network_type:
+        description: 'Network Type (mainnets or testnets)'
+        required: true
+        type: choice
+        options:
+        - mainnets
+        - testnets
+      network:
+        description: 'Network'
+        required: true
+        type: choice
+        options:
+        - hardhat
+        - bellecour
+jobs:
+  upgrade:
+    runs-on: ubuntu-latest
+    permissions:
+        contents: write # required by git-auto-commit-action
+    environment: ${{ inputs.network_type }} # Secrets for an environment are already limited to deployments on specific branches
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Setup Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: 22
+        cache: 'npm'
+    - name: Install dependencies
+      run: npm ci
+    - name: Set private key based on network
+      id: set-private-key
+      run: |
+        case "${{ inputs.network }}" in
+          "bellecour")
+            echo "PRIVATE_KEY=${{ secrets.BELLECOUR_PRIVATE_KEY }}" >> $GITHUB_ENV
```

```suggestion
            echo "PROD_PRIVATE_KEY=${{ secrets.BELLECOUR_PRIVATE_KEY }}" >> $GITHUB_ENV
```
and remove the line `PROD_PRIVATE_KEY: ${{ env.PRIVATE_KEY }}`